src/auth/googleauth.ts

```diff
@@ -634,4 +651,53 @@ export class GoogleAuth {
           if (callback) callback(null, body);
         });
   }
+
+
+  /**
+   * Returns the contents of the metadata of GC instance
+   * @return object representation of the service account JSON
+   */
+  getCredentials(
+      callback?: (err: Error|null, credentials?: CredentialBody) => void) {
+    this._checkIsGCE((err, gce) => {
+      if (gce) {
+        // For GCE, return the service account details from the metadata server
+        this.transporter.request(
+            {
+              method: 'GET',
+              uri:
+                  'http://metadata.google.internal/computeMetadata/v1/instance/service-accounts/?recursive=true',
+              headers: {'Metadata-Flavor': 'Google'}
+            },
+            (err, body, res) => {
+              if (err || !res || res.statusCode !== 200 || !body) {
+                return;
+              } else {
+                // Callback with the body
+                const credential:
+                    CredentialBody = {client_email: body['default']['email']};
+                if (callback) {
+                  callback(null, credential);
+                }
+              }
+            });
+      } else if (this.jsonContent) {
```

Comment:
Does this precedence make sense? Would it make more sense to check if user provided JSON, and return that first instead of trying to fetch it from GCE? @stephenplusplus might have some thoughts on this.

Reply:
Returned user provided JSON first.